### ISSUE TITLE
After Baseline Breathing, I received an error message 

### ISSUE THREAD

**Issue by choipaul92** [EmotionCognitionLab/CALM#24]
Ver 0.0.7

After completing the Baseline Breathing, I received an error message.
![Screenshot 2024-12-23 125246](https://github.com/user-attachments/assets/9a69cbef-097f-4e59-8a79-8fe6bd073a9b)

**Comment by noah10** [comment]
This is an internet latency problem. I'm not sure we're going to be able to avoid it entirely (though I have one idea that might make it happen a little less often), but when it does happen I can make sure that quitting a restarting the app will force the user to do the paced breathing sessions again so that we get another shot at calculating the hrv peaks.

**Comment by noah10** [comment]
I don't think that we're going to be able to prevent this from ever happening again, but after discussion with Mara I did make some changes to ameliorate the impact if it does happen. This happens when an error occurs in determining the peak power for a given frequency, which we have to do four times (once for the rest breathing and once for each of the three paced breathing measurements). Originally when that happened we forced the user to start over again. Now instead they will just have to re-do the measurement we didn't get a peak power for and all of the measurements after it. If we fail on the rest breathing then obviously this isn't an improvement, but if we fail on the third paced breathing it will be a significant difference.

I've tested this by adding some code that forces a failure when getting the peak power. Other than that, it's essentially impossible to test, so we'll just leave this open for a while and see if it happens again.

**Comment by choipaul92** [comment]
Ver: 1.2.1

After the baseline measurements were completed (resting baseline + 3-paced breathing baseline), I received an error message. However, after re-doing the baselines, the app progressed to cognitive assessments.

**Comment by ArielDRamirez24** [comment]
Ver: 1.2.2

I did not receive an error message. Cognitive exercises were up next after resting baseline and three paced breathing baselines.
